### ISSUE TITLE
PubSub: google.api_core.exceptions.ResourceExhausted: 429 Received message larger than max (28100227 vs. 4194304)

### ISSUE THREAD

**Issue by kpurdon** [googleapis/google-cloud-python#10056]
It seems this issue has been reported in fixed in a number of other places. I would expect the max size to be 10MB (as documented here: https://cloud.google.com/pubsub/quotas#resource_limits).

Related:

#5574
#5819
#5787

#### Environment details

- MacOS 10.15.2
- Cloud PubSub Emulator via `google/cloud-sdk:272.0.0` docker image

```
google-api-core==1.14.3
google-auth==1.8.2
google-cloud-core==1.1.0
google-cloud-pubsub==1.0.2
google-cloud-storage==1.23.0
google-resumable-media==0.5.0
googleapis-common-protos==1.6.0
grpc-google-iam-v1==0.12.3
```

**Comment by HemangChothani** [comment]
@kpurdon It's working fine for the service accounts (live) with max size 10MB, but the  emulator has some limitations and found suggestion of user on [google cloud platform](https://googlecloudplatform.uservoice.com/forums/302631-cloud-pub-sub/suggestions/37905820-emulator-needs-to-increase-max-grpc-message-size-t) forums.
@crwilcox  Could you comment on this?

**Comment by busunkim96** [comment]
@anguillanneuf @plamut Could you take a look at this issue?

**Comment by plamut** [comment]
@busunkim96 I can, although not in the immediate future (a lot of things on my hands ATM).

If I understand correctly, the emulator settings need to be updated to match the production's?

**Comment by HemangChothani** [comment]
@plamut Yes, that's correct, the emulator settings need to be updated to match the production.
